core/src/autogluon/core/trainer/abstract_trainer.py

```diff
@@ -457,12 +507,55 @@ def train_multi_levels(
                 additional_full_weighted_ensemble=additional_full_weighted_ensemble,
             )
             model_names_fit += base_model_names + aux_models
-        if self.model_best is None and len(model_names_fit) != 0:
+        if (self.model_best is None or infer_limit is not None) and len(model_names_fit) != 0:
```

Comment:
qq: could you explain this check ?

Reply:
Pseudocode breakdown of the check:

```
if (trainer has not specified which model is best) and (any model exists):
    select the best valid model based on validation score
elif (user specified infer limit) and (any model exists):
    select the best valid model based on validation score that satisfies the infer limit
```

The added infer_limit check ensures that we always return a model that satisfies the infer_limit constraints. Sometimes, a non-ensemble model can get a better score while satisfying infer limit than an ensemble model, and by default only ensemble models are set to `self.model_best` prior to this call. This call ensures that the best model will be picked, even if it isn't an ensemble.